runtime/queries/query/injections.scm

```diff
@@ -0,0 +1,6 @@
+((predicate 
+  name: (identifier) @_name
+  parameters: (parameters (string) @regex))
+ (#match? @_name "^#?(not-)?(match|vim-match|lua-match)$"))
```

Comment:
does "regex" lang also handle lua patterns (i e `lua-match` here) reasonably well?

Reply:
I don't know, I've never used that parser. That is precisely why I asked about dropping injections for parsers that are not bundled.

Reply:
it would still be a more conservative estimate that `regex` parser at least handles regexes, if not regexes + lua patterns (which are significantly more different than say, pcre regex vs vim regex)

Reply:
Oh, for sure. I just don't have the wherewithal to decide that. Either we trust nvim-treesitter and include this, or we don't ad I remove the whole injection.

Reply:
so I am supposed to file a PR at nvim-treesitter first to have them fix their broken shit? gotcha.

Reply:
not saying it's broken; just saying I don't know either way.

Reply:
I am saying it is broken. because it is broken.

Reply:
You might very well say so; I couldn't possibly comment.

Reply:
https://github.com/nvim-treesitter/nvim-treesitter/pull/3578/files#diff-e3a3db8fea6d634cefd718c9b57c3ffea1d6d9cadcace8ec1f7cb752cc5fa4a9

Reply:
Probably better to drop this injection for now.

Reply:
I think we should drop this from lua-match but maybe not vim-match.

Reply:
If you use `{vim-}match` in a query, you've already lost ;) (performance, that is)

Reply:
Lua Patterns (luap) syntactically are not the same as regex. That being said, this seems like a worthwhile problem to solve so I'll see about adding tree-sitter-luap to nvim-treesitter then adding a separate injection specifically for lua-match strings...

Reply:
There's already an open PR for nvim-treesitter that does exactly that...

Reply:
> There's already an open PR for nvim-treesitter that does exactly that...

Yeah just saw that sorry, I'll test it out now

Reply:
For anyone interested, https://github.com/nvim-treesitter/nvim-treesitter/pull/4433